extensions/warp-ipfs/src/store/message/community_task.rs

```diff
@@ -1275,6 +1283,105 @@ impl CommunityTask {
 
         Ok(())
     }
+    async fn process_join_event(&mut self, msg: Message) -> Result<(), Error> {
+        let data = PayloadMessage::<Vec<u8>>::from_bytes(&msg.data)?;
+        let community_id = self.community_id;
+        let sender = data.sender().to_did()?;
+        let event = serde_json::from_slice::<CommunityJoinEvents>(data.message()).map_err(|e| {
+            tracing::warn!(community_id = %community_id, sender = %data.sender(), error = %e, "Failed to deserialize message");
+            e
+        })?;
+        match event {
+            CommunityJoinEvents::Join => {
+                let mut is_invited = false;
+                for (_, invite) in &self.document.invites {
+                    if let Some(expiry) = invite.expiry {
+                        if expiry < Utc::now() {
```

Comment:
I wonder if we should move `Utc::now` into its own variable outside of the loop and use that when checking to see if it expired that way if there is a large subset of invites, it doesnt expire while it is processing, unless that is the intent. If so, then we probably should probably have a function apart of the invite that checks to see if it expired.

Reply:
done